### PR TITLE
add integration testing guide

### PR DESCRIPTION
**Pull request by crimsonfaith91**
Added a `.md` guide for integration testing.

### CODE REVIEW

**Comment by crimsonfaith91** [comment]
@droot Will you have time to review this guide?

File: integration.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+**Writing and Running Integration Tests**
+
+This article explores steps to write and run integration tests for Kubebuilder.
+
+Based on an example controller that watches *Parent* objects that in turn create *Child* objects, Kubebuilder provides framework for writing integration tests under `pkg/controller/parent/parent_controller_test.go`. Below is a detailed example of the file:
+```
```
Comment (droot) [inline integration.md]: We can improve the clarity if we use some *real* resources for parent and child. Consider using `ReplicaSet` as Parent and `Pod` as child.

File: integration.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+	// Verify Child is created.
+	child := &childv1alpha1.Child{}
+	g.Eventually(func() error { return c.Get(context.TODO(), childKey, child) }, timeout).
+		Should(gomega.Succeed())
+
+	// Manually delete Child since GC isn't enabled in the test control plane.
+	g.Expect(c.Delete(context.TODO(), child)).To(gomega.Succeed())
+}
+```
```
Comment (droot) [inline integration.md]: Consider removing boilerplate stuff such as import statements. Highlight the stuff which user is tied to reconcile logic and how to test that.

File: integration.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
+**Writing and Running Integration Tests**
+
+This article explores steps to write and run integration tests for Kubebuilder.
+
+Based on an example controller that watches *Parent* objects that in turn create *Child* objects, Kubebuilder provides framework for writing integration tests under `pkg/controller/parent/parent_controller_test.go`. Below is a detailed example of the file:
```
Comment (droot) [inline integration.md]: Also, one paragraph explaining what the test does. For. ex. how setup function brings up an APIServer and Etcd. Manager is started as part of the test itself. I have seen users struggling with it.

File: integration.md (hunk 2 above)
Comment (crimsonfaith91) [reply integration.md]: I included them to ensure what does `parentapis`, `parentv1alpha1`, `childapis`, and `childv1alpha1` means. I will remove all other import statements except for these 4.

File: integration.md (hunk 1 above)
Comment (droot) [reply integration.md]: I understand all of that you said above. Using some concrete resource examples are bit easier to understand. Also, I realized that when it comes to testing, testing core resources should not be any different from testing CRD resources :)

Don't have any strong opinion, so lets go ahead with the merge.

**Comment by droot** [comment]
Looks all good (except that we should move it under `docs` dir), pl. go ahead and squash the commits. Thanks.

File: integration.md (hunk 1 above)
Comment (crimsonfaith91) [reply integration.md]: > testing core resources should not be any different from testing CRD resources
Agreed, but we should also think what is the focus of kubebuilder, which (in my opinion) is about reconciliating custom resources, not core resources. Just my 2 cents. :D

**Comment by droot** [comment]
Thanks @crimsonfaith91 for the PR.

**Pull request merged**